proxy/ptunnel/commands/ssh.go

```diff
@@ -135,18 +172,29 @@ func NewSSH(base *client.BaseFlags) *SSH {
   $ ... ssh --proxy=https://gw.corp.enfabrica.net -- -p2222 user@10.10.0.12
 	Use a non default proxy to connect to your corp host.
 
+	$ ... ssh --proxy-map=.mtv=https://gw.corp.enfabrica.net,.foo=https://example.com -- p2222 user@machine-name.foo
+	Select from a few proxies based on string matching in the target address.
+	In this case https://example.com will be selected as the proxy, as the
+	target contains the string '.foo'.
+
   $ ... ssh --tunnel-extra="--browser-write-timeout=10s" -e "/bin/openssh" -- -p2222 user@10.10.0.12
 	Pass some extra flags to the tunnel command, use a different ssh than the one that can
 	be found in your path.
 `,
 		},
 		BaseFlags: base,
 	}
+	root.PreRunE = root.parseFlags
 	root.RunE = root.Run
 
 	root.Command.Flags().IntVar(&root.BufferSize, "buffer-size", 0, "Default read and write buffer size for window management. If 0, leave the default used by the tunnel")
 	root.Command.Flags().StringVarP(&root.Proxy, "proxy", "p", "", "Full url of the proxy to connect to, must be specified")
-
+	root.Command.Flags().StringSliceVar(
+		&root.proxyList,
+		"proxy-map",
+		nil,
+		"Map of suffix=gateway pairs to use for choosing proxy. Overrides --proxy when an entry matches any of the targets. If set, --proxy is used as the default fallback when no entries match. Entries should be non-overlapping/able to be applied in any order.",
```

Comment:
"Entries should be non-overlapping/able to be applied in any order."

is easy to miss, and the use of `map` above can cause problems for a subset of users that are hard to debug / pinpoint, as it will fail sometimes but work others (most likely would show up as a transient "ACL" / DNS resolution error?).

I have a slight preference to use a [] of struct above, and at least have consistent result/failure. Rest of the change looks great.

Reply:
Agreed; fixed.